poppy/fresnel.py

```diff
@@ -958,6 +959,52 @@ def apply_lens_power(self, optic, ignore_wavefront=False):
         _log.debug("------ Optic: " + str(optic.name) + " applied ------")
 
 
+    def _resample_wavefront_pixelscale(self, detector):
+        """ Resample a Fresnel wavefront to a desired detector sampling.
+
+        Parameters
+        ----------
+        detector : Detector class instance
+            Detector that defines the desired pixel scale
+
+        Returns
+        -------
+        The wavefront object is modified to have the appropriate pixel scale and spatial extent.
+
+        """
+        import scipy.ndimage
+
+        if self.angular_coordinates:
+            raise NotImplementedError("Resampling to detector doesn't yet work in angular coordinates for Fresnel.")
+
+        pixscale_ratio = (self.pixelscale / detector.pixelscale).decompose().value
+        _log.info("Resampling wavefront to detector with {} pixels and {}. Zoom factor is {}".format(
+            detector.shape, detector.pixelscale, pixscale_ratio ))
+
+        _log.debug("Wavefront pixel scale: {}".format(self.pixelscale))
+        _log.debug("Wavefront FOV: {} pixels, {}".format(self.shape, self.shape[0]*u.pixel*self.pixelscale))
+
+        _log.debug("Desired detector pixel scale: {}".format(detector.pixelscale))
+        _log.debug("Desired detector FOV: {} pixels, {}".format(detector.shape,
+                                                                      detector.shape[0]*u.pixel*detector.pixelscale,
+                                                                      ))
+
+        # TODO the following works but is not optimal for performance
+        # We should consider cropping out an appropriate subregion prior to performing the zoom.
+        # That makes a difference if the detector is only sampling a small part of a much larger wavefront
+
+        new_wf_real = scipy.ndimage.zoom(self.wavefront.real, pixscale_ratio)
```

Comment:
```suggestion
        new_wf_real = scipy.ndimage.zoom(self.wavefront.real, pixscale_ratio, order=self.interp_order)
```
Configuration of the interpolation order is useful for special cases, such where you want to keep masks binary

Reply:
see https://github.com/spacetelescope/poppy/blob/e97c97f01a002a71c86b07b1ce2eb8d473575f84/poppy/poppy_core.py#L1924

Reply:
Turns out this needs to be `order=detector.interp_order` rather than `self`. But yes in any case good suggestion.